packages/apps/marmalade-marketplace/src/components/Sale/DutchAuction.tsx

```diff
@@ -0,0 +1,286 @@
+    const currentPrice = auctionDetails ? getCurrentPrice({
+      startDate: Number((auctionDetails?.["start-date"] as unknown as IPactInt).int) * 1000,
+      endDate: Number((auctionDetails?.["end-date"] as unknown as IPactInt).int) * 1000,
+      startPrice: auctionDetails?.["start-price"],
+      reservePrice: auctionDetails?.["reserve-price"],
+      priceIntervalSeconds: Number((auctionDetails?.["price-interval-seconds"] as unknown as IPactInt).int)
+    }, new Date().getTime()) : 0;
+
+    setCurrentPrice(currentPrice)
+  }
+
+  useEffect(() => {
+    async function fetch() {
+
+      const auctionDetails = await getAuctionDetails({
+        auctionConfig: {
+          dutch: true
+        },
+        saleId: sale.saleId,
+        chainId: sale.chainId,
+        networkId: env.NETWORK_NAME,
+        host: env.CHAINWEB_API_HOST
+      });
+
+      console.log("auctionDetails", auctionDetails);
+
+      setAuctionDetails(auctionDetails)
+
+      updateCurrentPrice(auctionDetails);
+
+      const priceIntervalSeconds = Number((auctionDetails?.["price-interval-seconds"] as unknown as IPactInt).int) * 1000;
+      setNextPriceChange(getTimeUntilNextPriceChange(priceIntervalSeconds, new Date().getTime()));
+    }
+
+    fetch();
+  }, [])
+
+  useEffect(() => {
+
+    if (!auctionDetails) return;
+
+    if (isEndDateInPast) return;
+
+    const priceIntervalSeconds = Number((auctionDetails?.["price-interval-seconds"] as unknown as IPactInt).int) * 1000;
+
+    const currentPriceInterval = setInterval(() => updateCurrentPrice(auctionDetails), priceIntervalSeconds);
+    const priceDropInterval = setInterval(() => {
+      setNextPriceChange(getTimeUntilNextPriceChange(priceIntervalSeconds, new Date().getTime()));
+    }, 1000);
+
+    return () => {
+      clearInterval(currentPriceInterval);
+      clearInterval(priceDropInterval);
+    }
+
+  }, [auctionDetails, isEndDateInPast]);
+
+  const handleWithdraw = async () => {
+
+    if (!auctionDetails) return;
+
+    if (!isSellerConnected) {
+      alert("Only the seller can initiate withdrawal.");
+      return;
+    }
+
+    try {
+      await withdrawToken({
+        tokenId: auctionDetails?.["token-id"],
+        saleId: sale.saleId,
+        amount: new PactNumber(sale.amount).toPactDecimal(),
+        chainId: sale.chainId,
+        seller: {
+          account: sale.seller.account,
+          keyset: sale.seller.guard!
+        },
+        timeout: new PactNumber(sale.timeoutAt / 1000).toPactInteger()
+      }, {
+        ...config,
+        "defaults": { "networkId": config.networkId, meta: { "chainId": sale.chainId } }
+      }).execute();
+
+    } catch (error) {
+      alert("Error withdrawing token")
+      console.error(error);
+    }
+  }
+
+  const handleBuyNow = async () => {
+    if (!auctionDetails) return;
+
+    if (!account?.alias) {
+      alert("Please connect your wallet first to buy.");
+      return;
+    }
+
+
+    const FUNDING_ACCOUNT = "sender00";
```

Comment:
Do we need / want this private / public key info in the code?

Reply:
Sender00 is a keyset in a public repo (https://github.com/kadena-io/chainweb-node/blob/f258a705e77ce28c08cf934980c7d12cb4f5be30/pact/genesis/devnet/keys.yaml#L4) and is for the devnet admin key so I think this shouldn't be a problem. We'll need to change that to the buyer key in the future.